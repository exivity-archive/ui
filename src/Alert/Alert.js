import PropTypes from 'prop-types'
import styled, { css } from 'styled-components'
import { themed } from '../theme'

const styles = props => css`
  background-color: ${props.theme.colours.warning};
  padding: ${props.theme.spacing * props.theme.size}px;
  border-radius: ${props.theme.border.radius};

  ${props.danger && `
    color: ${props.theme.colours.bg};
    background-color: ${props.theme.colours.danger};
  `}
`

const Alert = styled.div`${styles}`

Alert.propTypes = {
  children: PropTypes.node.isRequired,
  danger: PropTypes.bool
}

export default themed(Alert)
