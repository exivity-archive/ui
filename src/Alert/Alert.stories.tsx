import React from 'react'
import { storiesOf } from '@storybook/react'
import { markdown } from '../utils/stories/markdown'
import { Alert } from '.'

storiesOf('atoms|Alert', module)
  .add('overview', markdown(require('./README.md')))
  .add('default', () => <Alert>Something is going on</Alert>)
  .add('success', () => <Alert success>All is well</Alert>)
  .add('danger', () => <Alert danger>Things went south</Alert>)
