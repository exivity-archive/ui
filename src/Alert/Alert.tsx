import styled from 'styled-components'
import { fromTheme, globalFont, matchThemeProp, StyledProps } from '../utils/styled'
import { preciseEm } from '../utils/styled/isolated'

export interface AlertProps extends StyledProps {
  // Purposes
  primary?: boolean
  secondary?: boolean
  success?: boolean
  danger?: boolean
}

export const Alert = styled.div.attrs({
  as: 'div',
  color: 'white'
})<AlertProps>`
  ${globalFont};

  background-color: ${matchThemeProp(theme => theme.global.purposes)};
  color: ${fromTheme(theme => theme.colors.white)};
  padding: ${preciseEm(0.5)}em ${preciseEm(1)}em;
  border: none;
  border-radius: ${fromTheme(theme => theme.global.borderRadius)}px;
`

Alert.displayName = 'Alert'
