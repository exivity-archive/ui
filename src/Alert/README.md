# Alert

Display a message with emphasis.

## Usage

```jsx
<Alert>Grab some attention</Alert>
```

### Success

```jsx
<Alert success>I'm positive</Alert>
```

### Danger

```jsx
<Alert danger>There was some sort of error</Alert>
```

## Props

<!-- props(Alert) -->
