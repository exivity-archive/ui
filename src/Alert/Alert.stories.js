import React from 'react'
import { storiesOf } from '@storybook/react'

import Alert from './Alert'

storiesOf('atoms|Alert', module)
  .add('default', () => <Alert>Could not load all requested information at this time, please try again later.</Alert>)
  .add('danger', () => <Alert danger>Something went utterly wrong while loading your report.</Alert>)
  .add('success', () => <Alert success>Synchronized all your data.</Alert>)
