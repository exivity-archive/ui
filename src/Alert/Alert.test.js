/* eslint-disable no-undef */
import React from 'react'
import renderer from 'react-test-renderer'

import Alert from './Alert'

test('renders default alert', () => {
  const alert = renderer.create(<Alert />)
  expect(alert.toJSON()).toMatchSnapshot()
})

test('renders dangerous alert', () => {
  const alert = renderer.create(<Alert danger />)
  expect(alert.toJSON()).toMatchSnapshot()
})
