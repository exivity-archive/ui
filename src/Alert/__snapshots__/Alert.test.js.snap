// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders dangerous alert 1`] = `
.c0 {
  background-color: #ffca71;
  padding: 16px;
  border-radius: 3px;
  color: #ffffff;
  background-color: #c33c32;
}

<div
  className="c0"
>
  danger
</div>
`;

exports[`renders default alert 1`] = `
.c0 {
  background-color: #ffca71;
  padding: 16px;
  border-radius: 3px;
}

<div
  className="c0"
>
  warning
</div>
`;
